### ISSUE TITLE
Assigning an image to a menu item created from an svg via wxBitmapBundle has a black background

### ISSUE THREAD

**Issue by dconnet** [wxWidgets/wxWidgets#22059]
I tracked the problem down to Commit d943d6f6

The addition of this call "AllocExclusive()" causes images created from an SVG file to have a black background when assigned to a menu. This images look correct for disabled menu items and toolbars. It appears to be just enabled menu items with this problem.

**Platform and version information**
 - wxWidgets version you use: dev
 - wxWidgets port you use: wxMSW
 - OS and its version: Windows 10.0.19044.1469

**Comment by vadz** [comment]
Thanks, I didn't have time to reply on the mailing list yet, but I'll try to look at this a.s.a.p.

As a temporary workaround we could at least check if the scale factor actually changes before doing this, because I think that in your case it doesn't (i.e. 1 before and after).

**Comment by dconnet** [comment]
One thing I forgot to mention: I'm currently running at 150% (main monitor).

**Comment by vadz** [comment]
Actually it just seems there is something wrong with alpha in menu images, i.e. with or without `AllocExclusive()`, the image doesn't appear correctly in the menu sample with this patch:
```diff
diff --git a/samples/menu/menu.cpp b/samples/menu/menu.cpp
index 6387705fc4..5b2cd27b30 100644
--- a/samples/menu/menu.cpp
+++ b/samples/menu/menu.cpp
@@ -32,6 +32,7 @@
     #include "wx/textdlg.h"
 #endif
 
+#include "wx/artprov.h"
 #include "wx/filehistory.h"
 #include "wx/filename.h"
 
@@ -559,7 +560,7 @@ bool MyApp::OnInit()
 #if USE_LOG_WINDOW
     wxMenuItem *item = new wxMenuItem(fileMenu, Menu_File_ClearLog,
                                       "Clear &log\tCtrl-L");
-    item->SetBitmap(copy_xpm);
+    item->SetBitmap(wxArtProvider::GetBitmap(wxART_COPY, wxART_MENU));
     fileMenu->Append(item);
     fileMenu->AppendSeparator();
 #endif // USE_LOG_WINDOW
```

It doesn't have black background, but I think it's another manifestation of the same problem. But, to be sure, could you please attach the SVG which you use? TIA!

**Comment by dconnet** [comment]
When I included that image (I bin2c'd it and pasted it directly into menu.cpp) and loaded it with:

> auto size = wxArtProvider::GetNativeSizeHint(wxART_MENU);
> wxBitmapBundle bundle = wxBitmapBundle::FromSVG(toolbarCopy_svg, size);
> item->SetBitmap(bundle.GetBitmap(size));

I get the black background.

**Comment by vadz** [comment]
Yes, this is definitely due to not copying the bitmap correctly when unsharing it, I've fixed this in #22085 now, so this should be fixed when the other PR is merged.

Thanks for reporting!

**Comment by PBfordev** [comment]
I assume this is the same issue that can be seen in `wxStaticBitmap` in the artprov sample, which however affects only non-SVGs, i.e., XPM based bitmaps?

v3.1.5 on the top, the master on the bottom:
![wx-black-bmp](https://user-images.githubusercontent.com/12495521/152395985-79ba6886-8f0b-4cb1-a41c-c2aa76d3999d.png)

**Comment by vadz** [comment]
I've extracted this into a different issue because this one is still visible after the fixes in dbc8506, thanks for reporting it (although I believe it's the same thing @kosh543 already ran into when working on #22087).
